### PR TITLE
Allow to unregister enricher.

### PR DESCRIPTION
**Pull request by schneiel**

### CODE REVIEW

**Review by hannesbochmann — CHANGES_REQUESTED** [review]
The code can be simpler.
```
    public static function unregister($class, $priority = 0)
    {
        if (self::$objects[$priority][$class] ?? false) {
            unset(self::$objects[$priority][$class]);
        }
    }
```

**Comment by schneiel** [comment]
@hannesbochmann Thank you for your suggested improvement in the commentary above. I have changed the code accordingly.

Furthermore, I noticed that the priority handling did not work before. I believe the integrated solution is the fastest and it is also clean. 

If you want to test the code, here you can: https://onlinephp.io/c/47888.

File: Classes/EnricherRegistry.php [hunk 1]
```diff
@@ -14,36 +14,30 @@ class EnricherRegistry implements EnricherInterface
...
-        }
-
-        unset(self::$objects[$priority][$class]);
     }
 
     public function process(FrontendUser $user, array $context)
     {
-        foreach ($this->flatten(self::$objects) as $className) {
-            GeneralUtility::makeInstance($className)->process($user, $context);
+        foreach ($this->flatten(self::$objects) as $className => $value) {
```
Comment (hannesbochmann) [inline Classes/EnricherRegistry.php]: To avoid such an unused variable like $value I would suggest to put the class name in line 27 into the array.

**Review by hannesbochmann — CHANGES_REQUESTED** [review]

File: Classes/EnricherRegistry.php (hunk 1 above)
Comment (schneiel) [reply Classes/EnricherRegistry.php]: @hannesbochmann I made the change. Do you have any further suggestions or can we merge this?

**Pull request merged**